### PR TITLE
Fixes #59 - adds exception handling for credential problems

### LINKED ISSUE #59: startup script crashes
immediately after hitting enter after typing in my password.
looks like you need to create the directories if they don't exist.
```
Traceback (most recent call last):
  File "startup.py", line 12, in <module>
    with open('data/unlocked/credentials.json','w') as credentials:
IOError: [Errno 2] No such file or directory: 'data/unlocked/credentials.json'
Traceback (most recent call last):
  File "gmusic.py", line 1, in <module>
    from gmusic.core.Core import Core
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/__init__.py", line 4, in <module>
    from gmusic.core import *
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/core/__init__.py", line 3, in <module>
    from gmusic.core.Core import Core
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/core/Core.py", line 1, in <module>
    from gmusic.content.ContentHandler import ContentHandler
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/content/__init__.py", line 2, in <module>
    from gmusic.content.ContentHandler import ContentHandler
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/content/ContentHandler.py", line 3, in <module>
    from gmusic.content.GMusicClient import GMusicClient
  File "/home/lprestonsegoiii/Development/gmusic/gmusic/content/GMusicClient.py", line 2, in <module>
    from gmusicapi import Mobileclient
ImportError: No module named gmusicapi
```

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
So, unicode just sorta works on my laptop. I wonder if there is some setting not set, or maybe a different terminal you should be using? idk.

### CODE REVIEW

**Comment by etkirsch** [comment]
I pushed the fix into the PR into your gmusic repo this afternoon along with the GST stuff.

**Comment by NullVoxPopuli** [comment]
Do you have a commit for that? I'm curious what happened

**Comment by NullVoxPopuli** [comment]
This needs to be merged, as it's error handling?

**Comment by etkirsch** [comment]
The name led me to believe that you were in the middle of fixing #59 and somehow had unicode working

**Comment by NullVoxPopuli** [comment]
Yeah, the name could have been better. 

What I meant was that this fixes alll the possible credential issues.

Also, I noticed that Unicode works in gnome-terminal without me changing anything.

**Comment by etkirsch** [comment]
Gotcha. Will check in a bit

**Comment by NullVoxPopuli** [comment]
renamed

File: gmusic/content/GMusicClient.py [hunk 1]
```diff
@@ -14,8 +15,17 @@ def __init__(self, data_cache):
...
+            credentials = json.load(open('data/unlocked/credentials.json', 'r'))
+            result = self.client.login(credentials['username'], credentials['password'], mac)
+            if result == False:
+                print('\n\033[93mLogin failed.')
+                print('Please double check that data/unlocked/credentials.json has correct information.\033[m\n')
+                os._exit(1)
+        except:
+            print('\n\033[93mdata/unlocked/credentials.json is not valid.')
+            print('You may need to delete and regnerate the credentials file.\033[m\n')
+            exit(1)
```
Comment (NullVoxPopuli) [inline gmusic/content/GMusicClient.py]: not sure if all exits should be `os._exit(1)`

I guess if we aren't actively trying to evade an Exception, then it doesn't matter.

File: gmusic/content/GMusicClient.py [hunk 2]
```diff
@@ -14,8 +15,17 @@ def __init__(self, data_cache):
...
-        self.client.login(credentials['username'], credentials['password'], mac)
+        try:
+            credentials = json.load(open('data/unlocked/credentials.json', 'r'))
+            result = self.client.login(credentials['username'], credentials['password'], mac)
+            if result == False:
+                print('\n\033[93mLogin failed.')
+                print('Please double check that data/unlocked/credentials.json has correct information.\033[m\n')
+                os._exit(1)
+        except:
+            print('\n\033[93mdata/unlocked/credentials.json is not valid.')
```
Comment (NullVoxPopuli) [inline gmusic/content/GMusicClient.py]: this prints in yellow

**Pull request merged**